src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -556,10 +654,11 @@ UniValue icxsubmitexthtlc(const JSONRPCRequest& request) {
                         {
                             {"offerTx", RPCArg::Type::STR, RPCArg::Optional::NO, "txid of offer tx for which the htlc is"},
                             {"amount", RPCArg::Type::NUM, RPCArg::Optional::NO, "amount in htlc"},
+                            {"receiveAddress", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "address for receiving DFC tokens in case of EXT/BTC order type"},
                             {"htlcScriptAddress", RPCArg::Type::STR, RPCArg::Optional::NO, "script address of external htlc"},
                             {"hash", RPCArg::Type::STR, RPCArg::Optional::NO, "hash of seed used for the hash lock part"},
                             {"ownerPubkey", RPCArg::Type::STR, RPCArg::Optional::NO, "pubkey of the owner to which the funds are refunded if HTLC timeouts"},
```

Comment:
What's the point of `htlcScriptAddress` and `ownerPubkey` since they are unused?

Reply:
They are not unused, but they are not used in Defichain. Those data is necessary so that the other party can create BTC HTLC on his side in order to monitor it and claim it. So he needs: htlc script address, owner pubkey, his pubkey (he already has his own) and timeout values to recreate it.